lightning/src/ln/channelmanager.rs

```diff
@@ -2514,7 +2514,21 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 						}));
 					}
 				}
-				try_chan_entry!(self, chan.get_mut().update_add_htlc(&msg, pending_forward_info), channel_state, chan);
+        let create_onion_error_packet = |chan: &Channel<ChanSigner>, pending_forward_info: &PendingHTLCStatus| {
+          match pending_forward_info {
+            &PendingHTLCStatus::Forward(PendingHTLCInfo { ref incoming_shared_secret, .. }) => {
+              let chan_update = self.get_channel_update(chan).unwrap();
```

Comment:
Is it safe to unwrap this? Above there is a check on whether the result is `Ok`.

Reply:
Fixed here https://github.com/rust-bitcoin/rust-lightning/pull/577/commits/9924863ad882e33b8e9d756db882492db518135a